lib/binaries/chrome_xml.ts

```diff
@@ -0,0 +1,147 @@
+import * as semver from 'semver';
+
+import {Config} from '../config';
+
+import {BinaryUrl} from './binary';
+import {XmlConfigSource} from './config_source';
+
+export class ChromeXml extends XmlConfigSource {
+  constructor() {
+    super('chrome', Config.cdnUrls()['chrome']);
+  }
+
+  getUrl(version: string): Promise<BinaryUrl> {
+    if (version === 'latest') {
+      return this.getLatestChromeDriverVersion();
+    } else {
+      return this.getSpecificChromeDriverVersion(version);
+    }
+  }
+
+  /**
+   * Get a list of chrome drivers paths available for the configuration OS type and architecture.
+   */
+  getVersionList(): Promise<string[]> {
+    return this.getXml().then(xml => {
+      let versionPaths: string[] = [];
+      let osType = this.getOsTypeName();
+
+      for (let content of xml.ListBucketResult.Contents) {
+        let contentKey: string = content.Key[0];
+
+        // Filter for 32-bit devices, make sure x64 is not an option
+        if (this.osarch !== 'x64' && contentKey.includes('64')) {
+          continue;
+        }
+
+        // Filter for only the osType
+        if (contentKey.includes(osType)) {
+          versionPaths.push(contentKey);
+        }
+      }
+      return versionPaths;
+    });
+  }
+
+  /**
+   * Helper method, gets the ostype and gets the name used by the XML
+   */
+  getOsTypeName(): string {
+    // Get the os type name.
+    if (this.ostype === 'Darwin') {
+      return 'mac';
+    } else if (this.ostype === 'Windows_NT') {
+      return 'win';
+    } else {
+      return 'linux';
+    }
+  }
+
+  /**
+   * Gets the latest item from the XML.
+   */
+  private getLatestChromeDriverVersion(): Promise<BinaryUrl> {
+    return this.getVersionList().then(list => {
+      let chromedriverVersion = null;
+      let latest = '';
+      let latestVersion = '';
+      for (let item of list) {
+        // get a semantic version
+        let version = item.split('/')[0];
+        if (semver.valid(version) == null) {
+          version += '.0';
+          if (semver.valid(version) == null) {
+            continue;
```

Comment:
Again, I'm not a fan of `continue`, but if this turns out to be a pain to remove you can leave it.